src/illuminatio/cleaner.py

```diff
@@ -16,59 +22,86 @@ def __init__(self, core_api, apps_api, rbac_api, logger=None):
+        """
+        Deletes all ServiceAccounts in a list of namespaces matching the given cleanup policy
+        """
+        return self.delete_resource_with_cleanup_policy(namespaces, cleanup_policy,
+                                                        self.core_api.delete_collection_namespaced_service_account,
+                                                        "SA")
 
-    def clean_up_daemon_sets_in_namespaces(self, namespaces, cleanup_policy):
-        return self.delete_res_using(namespaces, cleanup_policy,
-                                     self.apps_api.delete_collection_namespaced_daemon_set, "DS")
+    def clean_up_daemon_sets_in_namespaces_with_cleanup_policy(self, namespaces, cleanup_policy):
+        """
+        Deletes all DaemonSets in a list of namespaces matching the given cleanup policy
+        """
+        return self.delete_resource_with_cleanup_policy(namespaces, cleanup_policy,
+                                                        self.apps_api.delete_collection_namespaced_daemon_set, "DS")
 
-    def clean_up_namespaces(self, cleanup_policy):
-        resps = []
+    def clean_up_namespaces_with_cleanup_policy(self, cleanup_policy):
+        """
+        Deletes all namespaces matching the given cleanup policy
+        """
+        responses = []
         namespaces = self.core_api.list_namespace(
             label_selector=labels_to_string({CLEANUP_LABEL: cleanup_policy})).items
         namespace_names = [n.metadata.name for n in namespaces]
         self.logger.info("Deleting namespaces %s with cleanup policy %s", str(namespace_names), cleanup_policy)
         for namespace in namespaces:
             resp = self.core_api.delete_namespace(namespace.metadata.name, propagation_policy="Background")
-            resps.append(resp)
+            responses.append(resp)
         while self.core_api.list_namespace(label_selector=labels_to_string({CLEANUP_LABEL: cleanup_policy})).items:
-            self.logger.debug("Waiting for namespaces {} to be deleted.".format(namespace_names))
+            self.logger.debug("Waiting for namespaces %s to be deleted.", namespace_names)
             time.sleep(2)
-        return resps
+        return responses
 
-    def delete_res_using(self, namespaces, cleanup_policy, method, res_name):
-        resps = []
+    def delete_resource_with_cleanup_policy(self, namespaces, cleanup_policy, method, resource_name):
+        """
+        Deletes all resources which match the given cleanup policy
+        """
+        responses = []
         for namespace in namespaces:
-            self.logger.info("Deleting " + res_name + "s in " + str(namespace)
+            self.logger.info("Deleting " + resource_name + "s in " + str(namespace)
                              + " with cleanup policy " + cleanup_policy)
             resp = method(namespace, label_selector=labels_to_string({CLEANUP_LABEL: cleanup_policy}))
```

Comment:
This should actually be wrapped in a `try` `except` ? Otherwise `illuminatio` could crash during run time.